src/site/content/en/blog/fast-playback-with-preload/index.md

```diff
@@ -0,0 +1,599 @@
+segments. Note that some middleware libraries such as [Google's Shaka
+Player][shaka-player], [JW Player][jw-player], and [Video.js][video-js] are
+built to handle this for you.
+
+```js
+<video id="video" controls></video>
+
+<script>
+  const mediaSource = new MediaSource();
+  video.src = URL.createObjectURL(mediaSource);
+  mediaSource.addEventListener('sourceopen', sourceOpen, { once: true });
+
+  function sourceOpen() {
+    URL.revokeObjectURL(video.src);
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp09.00.10.08"');
+
+    // Fetch beginning of the video by setting the Range HTTP request header.
+    fetch('file.webm', { headers: { range: 'bytes=0-567139' } })
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      sourceBuffer.appendBuffer(data);
+      sourceBuffer.addEventListener('updateend', updateEnd, { once: true });
+    });
+  }
+
+  function updateEnd() {
+    // Video is now ready to play!
+    const bufferedSeconds = video.buffered.end(0) - video.buffered.start(0);
+    console.log(`${bufferedSeconds} seconds of video are ready to play.`);
+
+    // Fetch the next segment of video when user starts playing the video.
+    video.addEventListener('playing', fetchNextSegment, { once: true });
+  }
+
+  function fetchNextSegment() {
+    fetch('file.webm', { headers: { range: 'bytes=567140-1196488' } })
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      const sourceBuffer = mediaSource.sourceBuffers[0];
+      sourceBuffer.appendBuffer(data);
+      // TODO: Fetch further segment and append it.
+    });
+  }
+</script>
+```
+
+### Considerations
+
+As you're now in control of the entire media buffering experience, I suggest you
+consider the device's battery level, the "Data-Saver Mode" user preference and
+network information when thinking about preloading.
+
+#### Battery awareness
+
+Please take into account the battery level of users' devices before thinking
+about preloading a video. This will preserve battery life when the power level
+is low.
+
+Disable preload or at least preload a lower resolution video when the
+device is running out of battery.
+
+```js
+if ('getBattery' in navigator) {
+  navigator.getBattery()
+  .then(battery => {
+    // If battery is charging or battery level is high enough
+    if (battery.charging || battery.level > 0.15) {
+      // TODO: Preload the first segment of a video.
+    }
+  });
+}
+```
+
+#### Detect "Data-Saver"
+
+Use the `Save-Data` client hint request header to deliver fast and light
+applications to users who have opted-in to "data savings" mode in their
+browser. By identifying this request header, your application can customize and
+deliver an optimized user experience to cost- and performance-constrained
+users.
+
+Learn more by reading our complete [Delivering Fast and Light
+Applications with Save-Data][fast-light] article.
```

Comment:
```suggestion
See [Delivering Fast and Light Applications with Save-Data][fast-light] to learn more.
```
I want to avoid describing it as "complete" in the event that people go to the guide and do not find the information they're looking for